tests/rptest/tests/license_upgrade_test.py

```diff
@@ -108,3 +108,40 @@ def test_basic_upgrade(self):
 
         # Install license
         assert self.admin.put_license(license).status_code == 200
+
+
+class UpgradeMigratingLicenseVersion(RedpandaTest):
+    """
+    Verify that the cluster can interpret licenses between versions
+    """
+    def __init__(self, test_context):
+        super(UpgradeMigratingLicenseVersion, self).__init__(test_context=test_context,
+                                                     num_brokers=3,
+                                                     si_settings=SISettings())
+        self.installer = self.redpanda._installer
+        self.admin = Admin(self.redpanda)
+
+    def setUp(self):
+        # 22.2.x is when license went live
+        self.installer.install(self.redpanda.nodes, (22, 2, 7))
+        super(UpgradeMigratingLicenseVersion, self).setUp()
+
+    @cluster(num_nodes=3, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_license_upgrade(self):
+        license = sample_license()
+        if license is None:
+            self.logger.info(
+                "Skipping test, REDPANDA_SAMPLE_LICENSE env var not found")
+            return
+
+        # Upload a license
+        self.admin.put_license(license)
+
+        # Update all nodes to newest version
```

Comment:
This will need a followup to fix the version combinations: this is assuming that upgrades from 22.2->HEAD is what is wanted (i.e. on `dev` it will be skipping 22.3.1)).